catalyst-gateway/bin/src/db/index/block/rbac509/mod.rs

```diff
@@ -202,13 +206,16 @@ impl Rbac509InsertQuery {
             cache_persistent_rbac_chain(catalyst_id.clone(), new_chain);
         }
 
+        context.remove_addresses(removed_addresses.clone());
```

Comment:
As far as I can see, the addresses are also removed in the `record_valid_registration` function, so we don't need to remove them here.